### PR TITLE
Added new features to KAN model

### LINKED ISSUE #76: Enhance KAN
New functionalities will be added to the NN KAN model. Those are:

- Custom learning rate scheduling.
- The optimizer now can be changed. By default it uses Adam.
- More logging info is gathered, like learning rate evolution during training. 
- Now is easier to add more parameters to optimize by optima in terms of implementation.

### PR DESCRIPTION
**Pull request by DavidRamosArchilla**
Closes #76

### CODE REVIEW

File: pyLOM/NN/architectures/kan.py [hunk 1]
```diff
@@ -82,61 +101,144 @@ def fit(
...
+                Examples:
+                - StepLR: {"step_size": int, "gamma": float}.
+                - ReduceLROnPlateau: {"mode": str, "factor": float, "patience": int}.
+                - OneCycleLR: {"anneal_strategy": str, "div_factor": float}.
+                (default: `{}`).
+            opti_kwargs (dict, optional): Additional keyword arguments to pass to the optimizer (default: `{}`).
             print_eval_rate (int, optional): The model will be evaluated every ``print_eval_rate`` epochs and the losses will be printed. If set to 0, nothing will be printed (default: ``2``).
             loss_fn (torch.nn.Module, optional): The loss function (default: ``nn.MSELoss()``).
             save_logs_path (str, optional): Path to save the training and evaluation losses (default: ``None``).
+            intro (bool, optional): Whether to print the training information (default: ``True``).
```
Comment (ArnauMiro) [inline pyLOM/NN/architectures/kan.py:142]: maybe we should call this option `verbose` ?

**Review by ArnauMiro — CHANGES_REQUESTED** [review]

File: pyLOM/NN/architectures/kan.py (hunk 1 above)
Comment (DavidRamosArchilla) [reply pyLOM/NN/architectures/kan.py:142]: okey, I'll change that on the `__init__` too

**Review by ArnauMiro — APPROVED** [review]
